libnavigation-core/src/main/java/com/mapbox/navigation/core/stops/ArrivalController.kt

```diff
@@ -0,0 +1,101 @@
+package com.mapbox.navigation.core.stops
+
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * When navigating to points of interest, you may want to control the arrival
+ * experience. This interface gives you options to change arrival via [MapboxNavigation.attachArrivalController]
+ */
+interface ArrivalController {
+
+    /**
+     * Override the options for your arrival callback.
+     */
+    fun arrivalOptions(): ArrivalOptions
+
+    /**
+     * Based on your [ArrivalOptions], this will be called as the next stop is approached.
+     *
+     * To manually move to the next leg, return false and call [MapboxNavigation.navigateToNextStop]
+     *
+     * @return true to automatically move to the next step, false to do it manually
+     */
+    fun onArrival(routeLegProgress: RouteLegProgress): Boolean
+}
+
+/**
+ * The default controller for arrival. This will move onto the next leg automatically
+ * if there is one.
+ */
+class AutoArrivalController : ArrivalController {
+
+    /**
+     * Default arrival options
+     */
+    override fun arrivalOptions(): ArrivalOptions = ArrivalOptions.Builder().build()
+
+    /**
+     * By default this will move onto the next step.
+     */
+    override fun onArrival(routeLegProgress: RouteLegProgress): Boolean {
+        return true
+    }
+}
+
+/**
+ * Choose when to be notified of arrival.
+ */
+data class ArrivalOptions(
+
+    /**
+     * When the next stop less than [arrivalInSeconds] away,
+     * Mapbox will call [ArrivalController.onArrival]
+     */
+    val arrivalInSeconds: Double?,
+
+    /**
+     * When the next stop less than [arrivalInMeters] away,
+     * Mapbox will call [ArrivalController.onArrival]
+     */
+    val arrivalInMeters: Double?
+) {
+    /**
+     * Build your arrival options.
+     */
+    class Builder {
+
+        private var arrivalInSeconds: Double? = 0.0
+        private var arrivalInMeters: Double? = null
```

Comment:
was skeptic making default values here, because we don't have any examples using multiple stops. any choices are random

Reply:
I pointed it out because in the legacy we 👀 that threshold for auto-incrementing the leg index 👀 https://github.com/mapbox/mapbox-navigation-android/blob/5c1175f118f44aa6543c218403bdc69ef26dbeb7/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteProcessorRunnable.kt#L107-L113

Reply:
ok will take legacy's value :)